labgrid/remote/client.py

```diff
@@ -1531,6 +1572,24 @@ def main():
     tmc_subparser.add_argument('action', choices=['info', 'values'])
     tmc_subparser.set_defaults(func=ClientSession.tmc_channel)
 
+    subparser = subparsers.add_parser('xlx', help="connect to a Xilinx Vivado hardware server")
+    subparser.set_defaults(func=lambda _: subparser.print_help())
+    xlx_subparsers = subparser.add_subparsers(
+        dest='subcommand',
+        title='available subcommands',
+        metavar="SUBCOMMAND",
+    )
+
+    xlx_subparser = xlx_subparsers.add_parser('xsdb', help="run XSDB")
```

Comment:
I guess this exports a gdb compatible interface, so something not xilinx specific like `gdb` would be better.

Reply:
This is a proprietary, Tcl-based shell; see also my other comment about the client interface architecture.